File: app/src/main/java/mx/leo/sample/easyrecycler/adapter/SampleHeaderAdapter.java
package mx.leo.sample.easyrecycler.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import mx.leo.easyrecycler.adapter.EasyHeaderAdapter;
import mx.leo.easyrecycler.viewholder.EasyViewHolder;
import mx.leo.sample.easyrecycler.R;
import mx.leo.sample.easyrecycler.model.Language;

public class SampleHeaderAdapter extends EasyHeaderAdapter<Language> {

    @NotNull @Override public SampleHeaderViewHolder createHeaderViewHolder(@Nullable ViewGroup parent) {
        return new SampleHeaderViewHolder(LayoutInflater.from(parent.getContext()).
                inflate(R.layout.header_layout,parent,false));
    }

    @NotNull @Override public SampleItemViewHolder createItemViewHolder(@Nullable ViewGroup parent) {
        return new SampleItemViewHolder(LayoutInflater.from(parent.getContext()).
                inflate(R.layout.sample_item_layout,parent,false));
    }

    @Override public void onBindHeaderViewHolder(@NotNull EasyViewHolder headerHolder) {
        ((SampleHeaderViewHolder)headerHolder).bindHeader();
    }

    @Override public void onBindItemViewHolder(@NotNull EasyViewHolder holder, Language s, int position) {
        ((SampleItemViewHolder) holder).bindItem(s);
    }
}

File: samplekotlin/src/main/java/brauliomendez/com/kotlin/adapter/SampleHeaderAdapter.kt
package brauliomendez.com.kotlin.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import brauliomendez.com.kotlin.R
import brauliomendez.com.kotlin.model.Language
import brauliomendez.com.kotlin.viewholder.SampleHeaderViewHolder
import brauliomendez.com.kotlin.viewholder.SampleItemViewHolder
import mx.leo.easyrecycler.adapter.EasyHeaderAdapter
import mx.leo.easyrecycler.viewholder.EasyHeaderViewHolder
import mx.leo.easyrecycler.viewholder.EasyItemViewHolder
import mx.leo.easyrecycler.viewholder.EasyViewHolder

/**
 * Created by Braulio on 25/12/2016.
 */
class SampleHeaderAdapter : EasyHeaderAdapter<Language>() {

    override fun createHeaderViewHolder(parent: ViewGroup?): EasyHeaderViewHolder {
       return SampleHeaderViewHolder(LayoutInflater.from(parent?.context).inflate(R.layout.header_layout, parent, false))
    }

    override fun onBindHeaderViewHolder(headerHolder: EasyViewHolder) {
        val header = headerHolder as SampleHeaderViewHolder
        header.bindHeader()
    }

    override fun createItemViewHolder(parent: ViewGroup?): EasyItemViewHolder {
        return SampleItemViewHolder(LayoutInflater.from(parent?.context).inflate(R.layout.item_layout, parent, false))
    }

    override fun onBindItemViewHolder(holder: EasyViewHolder, language: Language, position: Int) {
        val item = holder as SampleItemViewHolder
        item.bindItem(language)
    }
}